packages/components/src/interfaces/native/Text.ts

```diff
@@ -28,14 +28,41 @@ const ItalicMarkSchema = Type.Object(
   },
 )
 
-const LinkMarkSchema = Type.Object(
-  {
-    type: Type.Literal("link", { default: "link" }),
-    href: Type.String(),
-  },
-  {
-    title: "Hyperlink",
-  },
+const LinkMarkSchema = Type.Unsafe<{
+  type: "link"
+  attrs: {
+    target: "_blank" | "_self" | "_parent" | "_top"
+    href: string | null
+    rel: string
+    class: string
+  }
+}>(
+  Type.Object(
+    {
+      type: Type.Literal("link", { default: "link" }),
+      attrs: Type.Object({
+        target: Type.Union([
+          // NOTE: Taken with reference to
+          // https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target
+          // There's technically 1 more property - unfencedTop but we're disallowing
+          // because it allows embedded frames to traverse beyond its root
+          Type.Literal("_blank"),
+          Type.Literal("_self"),
+          Type.Literal("_parent"),
+          Type.Literal("_top"),
+        ]),
+        // NOTE: The href given by tiptap here
+        // https://github.com/ueberdosis/tiptap/blob/main/packages/extension-link/src/link.ts
+        // defaults to `null` href but href could also be passed as `undefined`
+        href: Type.Union([Type.String(), Type.Null()]),
+        rel: Type.String(),
+        class: Type.Union([Type.String(), Type.Null()]),
```

Comment:
issue: I don't think we should match the schema that Tiptap provides entirely, our schema is intended to be a subset of the schema because we only want to allow certain behaviours instead of all the available options that HTML/Tiptap provides. Adding all here means we lose the signal on what is allowed and what isn't.

In general we override the `rel` and `class` when we output to the user, so those properties will be ignored. For `target` we should only support `_blank`.

Reply:
done - where do i see the properties that we strip? couldn't find a `Text` component

Reply:
Mainly in getSanitizedInlineContent.ts:

https://github.com/opengovsg/isomer/blob/c19c5d5cfbc26f8a04ad5052727906ebda5df741/packages/components/src/utils/getSanitizedInlineContent.ts#L5-L42